File: src/0485.max-consecutive-ones.485/solution.ts
function findMaxConsecutiveOnes (nums: number[]): number {
    let result = 0;
    let index = 0;
    while (index < nums.length) {
        if (nums[index] === 0) {
            index++;
            continue;
        }
        let count = 0;
        while (index < nums.length && nums[index] === 1) {
            index++;
            count++;
        }
        result = Math.max(result, count);
    }
    return result;
}

File: src/0485.max-consecutive-ones.485/solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int result = 0;
        int index = 0;
        while (index<nums.length){
            if(nums[index++] == 0){
                continue;
            }
            int cnt = 1;
            while(index<nums.length && nums[index] == 1){
                cnt++;
                index++;
            }
            result = Math.max(result,cnt);
        }
        return result;
    }
}